### ISSUE TITLE
Can not map enums when using GraphQLFragment, no errors when building query

### ISSUE THREAD

**Issue by everscope** [byme8/ZeroQL#99]
**Describe the bug**

When using GraphQLFragment, it is unclear how Enums should be mapped
Also when doing this, there is no exception when sending or building query - query will be build, but won't select any fields that will lead to exception on GraphQL server.
Am I missing something or this is not possible?

**How to Reproduce**

Code example, simple query towards GraphQL api and mapping result to read model using fragments (see graphQL schema below)
```csharp
    public class Person 
    {
         ... other fileds ...
        public MyOwnStateEnum MyOwnEnumType { get; set; }
    }

    public enum MyOwnStateEnum 
    {
      STATE1
      STATE2
      ...
    }

    [GraphQLFragment]
    public static PersonReadModel ToPersonReadModel(
        this Person p) => new()
        {
              ... other fileds ...
              //this one does not work
              State = (MyOwnStateEnum)p.State,
              // this aslo does not work
              //State = (MyOwnStateEnum) Enum.Parse(typeof(MyOwnStateEnum), p.State.Tostring());,
        };


    var response = await _myGraphQlService
        .Query(variables, static (i, q) =>
            q.Persons(
                where: i.Where,
                selector: n => n.ToPersonReadModel()
            )
        );
```
Query: `query ($where: PersonFilterInput!) { persons(where: $where) { } } }` (query does not select any fields)
After removing Enum convertion everything works fine (all other fields are selected)
Then this query obviously returns error when running it (as you can not have query which is not selecting any fields)

**Expected behavior**
 
Ability to map enum and/or exception message saying that query is impossible to build
Today, if you use same query (with type casting) inside selector, it will be recognized as error.

**The GraphqQL schema that fails**

If applicable, there is no need to post the whole schema, but having the small part that causes the issue would be excellent.
```graphql
type Person {
  state: PersonStateDto
...other fields...
}

enum PersonStateDto {
  STATE1
  STATE2
  ...
  ...
}

```

**Environment (please complete the following information):**
 - Nuget package version [6.1.1]
 - IDE: [VS]
 - .Net Version [8]

**Comment by byme8** [comment]
I have tried to reproduce the issue with ZeroQL 6.1.1 and for me works fine 
![image](https://github.com/byme8/ZeroQL/assets/5927686/c6ad19dd-54c5-4f87-89e4-d8e1f0389375)

**Comment by byme8** [comment]
I am using Microsoft Visual Studio Enterprise 2022 (64-bit) Version 17.9.5

**Comment by everscope** [comment]
I am using Visual Studio Community 2022 17.9.6 64-bit

I checked again, it looks like it works perfectly fine if you have only one invalid cast
![image](https://github.com/byme8/ZeroQL/assets/79381961/11d98c33-210a-4984-8c05-253e571768b2)


but if you add another fields, it is not highlighted as error
![image](https://github.com/byme8/ZeroQL/assets/79381961/4490f86b-73ac-41d6-a12d-11cb2a5c66a0)


and same happens if I you have 2 casts: only first one will be counted as error
![image](https://github.com/byme8/ZeroQL/assets/79381961/db4fe0f6-fd29-4f8a-8ad0-4aee84543fa0)
![image](https://github.com/byme8/ZeroQL/assets/79381961/2b68bfed-6ca1-41f6-a368-2a43ec0064c1)



And do I understand correctly, there is no way to cast enums inside fragments, right?

**Comment by everscope** [comment]
Tested on Microsoft Visual Studio Enterprise 2022 (64-bit) (v. 17.9.6) with same result

**Comment by byme8** [comment]
>if you add another fields, it is not highlighted as error

I managed to reproduce this case. It should report the error.

> And do I understand correctly, there is no way to cast enums inside fragments, right?

The fragment here should not treated as a C# function. It is more like GraphQL query declaration powered by C# syntax. It is analyzed at compile time to bake everything into proper raw GraphQL query. 

At the moment, the source generator supports the limited C# syntax tree to simplify the implementation and improve performance.
I am think that it would be fine to support syntax like `` (UserKind2)user.UserKind ``. However `` Enum.Parse `` it is too much.

**Comment by byme8** [comment]
Checkout v6.2.1-preview.1.

The missing error message has been fixed in v6.2.0. The v6.2.1-preview.1 adds support for casting enums.
